management/examples/createVM/createVM.go

```diff
@@ -0,0 +1,87 @@
+package createVM
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+	"github.com/Azure/azure-sdk-for-go/management/hostedservice"
+	"github.com/Azure/azure-sdk-for-go/management/storageservice"
+	"github.com/Azure/azure-sdk-for-go/management/virtualmachine"
+	"github.com/Azure/azure-sdk-for-go/management/vmutils"
+)
+
+func createVirtualMachine(psPath string) {
+	dnsName := fmt.Sprintf("test-vm-from-go-%v", randomString(5))
+	storageAccount := fmt.Sprintf("stoacc%v", randomString(5))
+	location := "central us"
+	vmSize := "Small"
+	vmImage := "b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04-LTS-amd64-server-20140724-en-us-30GB"
+	userName := "testuser"
+	userPassword := "Test123"
+
+	fmt.Println("Create client")
+	client, err := management.ClientFromPublishSettingsFile(fmt.Sprintf("%v.publishsettings", psPath), "")
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+		return
+	}
+
+	fmt.Println("Create hosted service")
+	if err := hostedservice.NewClient(client).CreateHostedService(hostedservice.CreateHostedServiceParameters{
+		ServiceName: dnsName,
+		Location:    location,
+		Label:       base64.StdEncoding.EncodeToString([]byte(dnsName))}); err != nil {
+		log.Fatalf("Error: %v", err)
+		return
+	}
+
+	fmt.Println("Create storage account")
+	_, err = storageservice.NewClient(client).CreateStorageService(storageservice.StorageAccountCreateParameters{
+		ServiceName: storageAccount,
+		Label:       base64.URLEncoding.EncodeToString([]byte(storageAccount)),
+		Location:    location,
+		AccountType: storageservice.AccountTypeStandardLRS})
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+		return
+	}
+
+	fmt.Println("Create virtual machine")
+	role := vmutils.NewVMConfiguration(dnsName, vmSize)
+	vmutils.ConfigureDeploymentFromPlatformImage(
+		&role,
+		vmImage,
+		fmt.Sprintf("http://%s.blob.core.windows.net/%s/%s.vhd", storageAccount, dnsName, dnsName),
+		"")
+	vmutils.ConfigureForLinux(&role, dnsName, userName, userPassword)
+	vmutils.ConfigureWithPublicSSH(&role)
+
+	fmt.Println("Deploy")
+	operationID, err := virtualmachine.NewClient(client).
+		CreateDeployment(role, dnsName, virtualmachine.CreateDeploymentOptions{})
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+		return
+	}
+	fmt.Println("Waiting...")
+	if err := client.WaitForOperation(operationID, nil); err != nil {
+		log.Fatalf("Error: %v", err)
+		return
+	}
+}
+
+func randomString(n int) string {
```

Comment:
this random is not real. it will produce same number sequence on everybody's machine as golang default rand seed is `0` on everybody's machine.